.gitignore

```diff
@@ -20,6 +20,7 @@ kubedns
 # pilot
 pilot/pkg/kube/config
 pilot/pkg/proxy/envoy/envoy
+pilot/docker/pilot-discovery
```

Comment:
Why ? It would be a regression, we don't want binaries in the source tree!

Reply:
It's gitignore, why does it matter? Yes I have my own custom "regression" scripts, that are not part of this PR.

Reply:
It matters because we don't want to go back to having dangling binaries in the source. 
( we should remove the others now that they go to out )

Reply:
There's tons of stuff in this gitignore already that do not have to be there (kubedns?). Why are you nit-picking this little addition?